src/Faithlife.Analyzers/StringAnalyzer.cs

```diff
@@ -0,0 +1,57 @@
+using System.Collections.Immutable;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Operations;
+
+namespace Faithlife.Analyzers
+{
+	[DiagnosticAnalyzer(LanguageNames.CSharp)]
+	public sealed class StringAnalyzer : DiagnosticAnalyzer
+	{
+		public override void Initialize(AnalysisContext analysisContext)
+		{
+			analysisContext.RegisterCompilationStartAction(compilationStartAnalysisContext =>
+			{
+				compilationStartAnalysisContext.RegisterOperationBlockStartAction(context =>
+				{
+					context.RegisterOperationAction(oc => AnalyzeOperation(oc), OperationKind.InterpolatedString);
+				});
+			});
+		}
+
+		public const string DiagnosticId = "FL0007";
+
+		public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(s_rule);
+
+		private static void AnalyzeOperation(OperationAnalysisContext context)
+		{
+			var invocationOperation = (IInterpolatedStringOperation) context.Operation;
+			var foundDollarSign = false;
+
+			foreach (var child in invocationOperation.Children)
+			{
+				if ((child as IInterpolatedStringTextOperation)?.Text.Syntax.ToFullString().EndsWith("$") ?? false)
```

Comment:
This is a violation of https://github.com/Faithlife/FaithlifeAnalyzers/wiki/FL0002. Which must mean we haven't set up FaithlifeAnalyzers to analyze FaithlifeAnalyzers?